networkx/algorithms/shortest_paths/weighted.py

```diff
@@ -125,7 +125,18 @@ def dijkstra_path_length(G, source, target, weight='weight'):
     --------
     bidirectional_dijkstra()
     """
-    length = dict(single_source_dijkstra_path_length(G, source, weight=weight))
+    
+    if source == target:
+        return 0
+
+    if G.is_multigraph():
+        get_weight = lambda u, v, data: min(
+            eattr.get(weight, 1) for eattr in data.values())
+    else:
+        get_weight = lambda u, v, data: data.get(weight, 1)
+    
+    length =  dict(_dijkstra(G, source, get_weight, target=target))
```

Comment:
It seems we don't need this dict either.  We just want the one length. So we can iterate through the values from ```_dijkstra``` and look for the target. If it doesn't come we know to throw the exception. That saves the space for the dict and probably time too. Something like:

    for n,length in _dijkstra(G, source, get_weight, target=target)
        if n == target:
            return length
    raise nx.NetworkXNoPath("node %s not reachable from %s"%(source, target))

what do you think?